src/layouts/components/Homepage/AnnouncementBody.tsx

```diff
@@ -0,0 +1,224 @@
+import { Text, Box, FormControl, VStack } from "@chakra-ui/react"
+import { DragDropContext } from "@hello-pangea/dnd"
+import {
+  FormLabel,
+  Input,
+  FormErrorMessage,
+  Button,
+  DatePicker,
+  Textarea,
+} from "@opengovsg/design-system-react"
+import _ from "lodash"
+import { BiPlus } from "react-icons/bi"
+
+import { useEditableContext } from "contexts/EditableContext"
+
+import { Editable } from "layouts/components/Editable"
+
+import { AnnouncementError, AnnouncementOption } from "types/homepage"
+
+const MAX_ANNOUNCEMENTS = 5
+
+interface AnnouncementBodyProps {
+  errors: {
+    announcementItems: AnnouncementError[]
+  }
+  announcementItems: Partial<AnnouncementOption>[]
+}
+
+export const AnnouncementBody = ({
+  errors,
+  announcementItems = [],
+}: AnnouncementBodyProps) => {
+  const {
+    onDragEnd,
+    onChange,
+    onCreate,
+    onDelete,
+    onDisplay,
+  } = useEditableContext()
+  return (
+    <Box w="full">
+      <DragDropContext onDragEnd={onDragEnd}>
+        <Editable.Droppable editableId="announcement">
+          <VStack spacing="1.25rem" align="flex-start" p={0} />
+          <Text mt="1.5rem" textStyle="h6">
+            Announcements
+          </Text>
+          <Text mt="0.5rem" textStyle="body-2" textColor="base.content.medium">
+            You can display up to 5 announcements at a time. Newly added
+            announcements are shown on the top of the list
+          </Text>
+
+          <Editable.Accordion onChange={() => onDisplay("announcement")}>
+            <Editable.EmptySection
+              isEmpty={announcementItems.length === 0}
+              title="Announcements you add will appear here"
+              subtitle=""
+            >
+              <Editable.Section px={0} spacing="0.75rem" py="1.5rem">
+                {announcementItems.map(
+                  (
+                    {
+                      title: announcementTitle,
+                      date: announcementDate,
+                      announcement: announcementContent,
+                      link_text: announcementLinkText,
+                      link_url: announcementLinkUrl,
+                    },
```

Comment:
Not a functional comment but just wondering if it would be better to just have `announcement` object here, then call things like `announcement.title` below.

Reply:
hey i rather destructure at the top bah. 
i didnt want to do announcement.announcement for clarity